data-prepper-plugins/http-source/src/test/java/com/amazon/dataprepper/plugins/source/loghttp/HTTPSourceConfigTest.java

```diff
@@ -28,4 +38,82 @@ public void testDefault() {
+            file = new File(temporaryDirectory, UUID.randomUUID().toString());
+            file.createNewFile();
+        }
+
+        @Test
+        void isSslCertificateFileValidation_should_return_true_if_ssl_is_false() throws NoSuchFieldException, IllegalAccessException {
+            final HTTPSourceConfig objectUnderTest = new HTTPSourceConfig();
+
+            reflectivelySetField(objectUnderTest, "ssl", false);
+
+            assertThat(objectUnderTest.isSslCertificateFileValidation(), equalTo(true));
+        }
+
+        @Test
+        void isSslCertificateFileValidation_should_return_false_if_ssl_is_true_and_sslCertificateFile_is_null() throws NoSuchFieldException, IllegalAccessException {
+            final HTTPSourceConfig objectUnderTest = new HTTPSourceConfig();
+
+            reflectivelySetField(objectUnderTest, "ssl", true);
+
+            assertThat(objectUnderTest.isSslCertificateFileValidation(), equalTo(false));
+        }
+
+        @Test
+        void isSslCertificateFileValidation_should_return_false_if_ssl_is_true_and_sslCertificateFile_is_a_valid_file() throws NoSuchFieldException, IllegalAccessException {
+            final HTTPSourceConfig objectUnderTest = new HTTPSourceConfig();
+
+            reflectivelySetField(objectUnderTest, "ssl", true);
+            reflectivelySetField(objectUnderTest, "sslCertificateFile", file.getAbsolutePath());
+
+            assertThat(objectUnderTest.isSslCertificateFileValidation(), equalTo(true));
+        }
+
+        @Test
+        void isSslKeyFileValidation_should_return_true_if_ssl_is_false() throws NoSuchFieldException, IllegalAccessException {
+            final HTTPSourceConfig objectUnderTest = new HTTPSourceConfig();
+
+            reflectivelySetField(objectUnderTest, "ssl", false);
+
+            assertThat(objectUnderTest.isSslKeyFileValidation(), equalTo(true));
+        }
+
+        @Test
+        void isSslKeyFileValidation_should_return_true_if_ssl_is_false_and_sslKeyFile_is_null() throws NoSuchFieldException, IllegalAccessException {
+            final HTTPSourceConfig objectUnderTest = new HTTPSourceConfig();
+
+            reflectivelySetField(objectUnderTest, "ssl", true);
+
+            assertThat(objectUnderTest.isSslKeyFileValidation(), equalTo(false));
+        }
+        @Test
+        void isSslKeyFileValidation_should_return_true_if_ssl_is_false_and_sslKeyFile_is_a_valid_file() throws NoSuchFieldException, IllegalAccessException {
+            final HTTPSourceConfig objectUnderTest = new HTTPSourceConfig();
+
+            reflectivelySetField(objectUnderTest, "ssl", true);
+            reflectivelySetField(objectUnderTest, "sslKeyFile", file.getAbsolutePath());
+
+            assertThat(objectUnderTest.isSslKeyFileValidation(), equalTo(true));
+        }
+
+        private void reflectivelySetField(final HTTPSourceConfig httpSourceConfig, final String fieldName, final Object value) throws NoSuchFieldException, IllegalAccessException {
```

Comment:
Is there a reason you like reflectively setting the fields on HTTPSourceConfig better than creating a `Map<String, Object>` and then using Jackson to convert to HTTPSourceConfig? Will validation get skipped if you do this?

Reply:
I had no particular reason. Both approaches seemed about the same effort, so I just went with one.

Reply:
I do think this is a little more readable because each value set is quite intentional. Sometimes in tests there are objects put into Maps and as a reader you aren't quite sure if they are really needed or not.